obspy/signal/spectral_estimation.py

```diff
@@ -1285,6 +1297,191 @@ def _add_npz(self, filename):
+            plt.close()
+        elif show:
+            plt.draw()
+            plt.show()
+        else:
+            plt.draw()
+            return fig
+
+    def plot_temporal(self, period, color=None, legend=True, grid=True,
+                      linestyle="-", marker=None, filename=None, show=True,
+                      **temporal_restrictions):
+        """
+        Plot the evolution of PSD value of one (or more) period bins over time.
+
+        If a filename is specified the plot is saved to this file, otherwise
+        a matplotlib figure is returned or shown.
+
+        Additional keyword arguments are passed on to :meth:`_stack_selection`
+        to restrict at which times PSD values are selected (e.g. to compare
+        temporal evolution during a specific time span of each day).
+
+        .. note::
+            For example plots see the :ref:`Obspy Gallery <gallery>`.
+
+        :type period: float (or list thereof)
+        :param period: Period of PSD values to plot. The period bin with the
+            central period that is closest to the specified value is selected.
+            Multiple values can be specified in a list (``color`` option should
+            then also be a list of color specifications, or left ``None``).
+        :type color: matplotlib color specification (or list thereof)
+        :param color: Color specification understood by :mod:`matplotlib` (or a
+            list thereof in case of multiple periods to plot). ``None`` for
+            default colors.
+        :type grid: bool
+        :param grid: Enable/disable grid in histogram plot.
+        :type legend: bool
+        :param legend: Enable/disable grid in histogram plot.
+        :type linestyle: str
+        :param linestyle: Linestyle for lines in the plot (see
+            :func:`matplotlib.pyplot.plot`).
+        :type marker: str
+        :param marker: Marker for lines in the plot (see
+            :func:`matplotlib.pyplot.plot`).
+        :type filename: str
+        :param filename: Name of output file
+        :type show: bool
+        :param show: Enable/disable immediately showing the plot.
+        """
+        import matplotlib.pyplot as plt
+        from matplotlib.dates import date2num
+
+        try:
+            len(period)
+        except TypeError:
+            periods = [period]
+        else:
+            periods = period
+
+        if color is None:
+            colors = [None] * len(periods)
+        else:
+            if len(periods) == 1:
+                colors = [color]
+            else:
+                colors = color
+
+        period_bin_centers = self.period_bin_centers
+        times = date2num([t.datetime for t in self.times_processed])
+
+        if temporal_restrictions:
+            mask = ~self._stack_selection(**temporal_restrictions)
+        else:
+            mask = None
+
+        fig = plt.figure()
+        ax = fig.add_subplot(1, 1, 1)
+
+        for period, color in zip(periods, colors):
```

Comment:
I would love to have the `ppsd_extract_time_series(period)` as a static method outside of the imaging, could be useful for other purposes, in my case: getting the time series for all my stations and plotting a map of the noise level.

this would also allow testing the period extraction (currently not done, is it?, or done via-via the image comparison?)